### PR TITLE
Add Terraform docs/HA guide

### PR DESCRIPTION
**Pull request by webvictim**
Work in progress

### CODE REVIEW

**Comment by webvictim** [comment]
Should be ready to check now.

**Review by benarent — APPROVED** [review]
I pushed a few changes to your branch to speed up production cycle :shipit:

File: docs/4.2/aws_terraform_guide.md [hunk 1]
```diff
@@ -0,0 +1,793 @@
...
+## Reference deployment defaults
+
+### Instances
+
+Our reference deployment will provision the following instances for your cluster by default:
+
+- 2 x m4.large Teleport **auth** instances in an ASG, behind an internal network load balancer, configured using DynamoDB for
+shared storage. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/auth_asg.tf#L11)
+- 2 x m4.large Teleport **proxy** instances in an ASG, behind a public-facing load balancer - NLB for LetsEncrypt, ALB for ACM. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/proxy_asg.tf#L12)
+- 1 x m4.large Teleport **node** instance in an ASG. [The desired size of the ASG is configured here](https://github.com/gravitational/teleport/blob/master/examples/aws/terraform/node_asg.tf#L10)
```
[the remark this answers is not in the document]
Comment (webvictim) [reply docs/4.2/aws_terraform_guide.md]: I think we were basically trying to get something where it'd be good enough for pretty much anyone but the most hardcore users. m4.large is 10 cents an hour - I guess for 5 of those instances it adds up pretty quickly.

One thing I would like to do with Terraform is expose the instance types more readily as a variable which can be changed at runtime and make some recommendations for scale. It's just something that I haven't got around to testing yet.

**Comment by webvictim** [comment]
Thanks @benarent.

Incidentally this reminds me that we don't seem to have a page in the docs about how to use `tctl` remotely. It'd be cool if I could link to that from here to describe a way to run `tctl` commands without needing to SSH into the cluster.

File: docs/4.2.yaml [hunk 2]
```diff
@@ -21,6 +21,7 @@ markdown_extensions:
   - footnotes
   - toc:
       marker: '[TOC]'
+      toc_depth: 2
```
Comment (webvictim) [inline docs/4.2.yaml]: Love this, didn't realise we could do it.

Comment (benarent) [reply docs/4.2.yaml]: ^ Yeah, it' a new-ish feature... I've been a bit slow to adopt the new features after I updated the MkDocs image.

**Comment by benarent** [comment]
I would go ahead and merge this, I'll update a reference when I work on https://github.com/gravitational/teleport/issues/3018

**Pull request merged**